packages/core-ethereum/src/commitment.ts

```diff
@@ -57,14 +70,45 @@ async function createCommitmentChain(db: HoprDB, channelId: Hash, setChainCommit
   log('commitment chain initialized')
 }
 
+/**
+ * Simple class encapsulating channel information
+ * used to generate the initial channel commitment.
+ */
+export class ChannelCommitmentInfo {
+  constructor(
+    public readonly channelEntry: ChannelEntry,
+    public readonly chainId: number,
+    public readonly contractAddress: string
+  ) {}
+
+  /**
+   * Generate the initial commitment seed using this channel information and the given
+   * private node key.
+   * @param nodePrivateKey Local node private key.
+   */
+  public createInitialCommitmentSeed(nodePrivateKey: Uint8Array): Uint8Array {
+    const channelSeedInfo = u8aConcat(
+      this.channelEntry.channelEpoch.serialize(),
+      toU8a(this.chainId, 4),
+      this.channelEntry.getId().serialize(),
+      toUtf8Bytes(this.contractAddress)
+    )
+
+    return deriveCommitmentSeed(nodePrivateKey, channelSeedInfo)
+  }
+}
+
 export async function initializeCommitment(
   db: HoprDB,
-  channelId: Hash,
+  nodePrivateKey: Uint8Array,
+  channelInfo: ChannelCommitmentInfo,
```

Comment:
Why don't take a `ChannelEntry` instance and chain information (contract address, chainId) and derive all other information?

In the current version the caller needs to know a bunch of internal information which can hide in this class.

Reply:
That was my first thought. But then I was more like 
- what if there will be more information needed for initial commitment at some point
- if it happens that these will need to be collected at several places in code, it's simpler to pass reference to this class and collect them in it
- I don't like functions that take more than 5 args

So these kinda out-weighted the drawback for me of having to export an additional class